### PR TITLE
docs: move docs to docusaurus

### PR DESCRIPTION
**Pull request by paularmstrong**
# Live preview: https://frigate-nzwqyym4t.vercel.app/

## Motivation
Frigate has a fairly complex set of instructions for configuration with many options, requirements, and advanced features. It is also moving fairly quickly with some pretty important breaking changes, which makes it hard to find the correct documentation for the version that someone is using.

## Solution
Use a documentation site generated by [Docusaurus](https://v2.docusaurus.io). 

* Can be built and automatically deployed to `gh-pages`, hosted for free by GitHub. 
* Supports [versioned documentation](https://v2.docusaurus.io/docs/versioning)
* [Search](https://v2.docusaurus.io/docs/search)
* Automatically hashes builds so no stale cache issues

This patch just sets up the basics:
* Ran `npx @docusaurus/init@latest init docs classic`
* Moved a few images around
* Copy/pasted some of the `README.md` file into various pages to show how to set things up (see: `docs/docs/*/**.md` files)

Screen Shot 2021-01-20 at 4 38 06 PM

Screen Shot 2021-01-20 at 4 38 22 PM

### Testing:
* Ensure you have node 12 or 14 installed
* `cd docs`
* `npm install`
* `npm run start` (your browser window should automatically open to `http://localhost:3000`)
* Edit any file in docs and the browser will auto-refresh

### In order to complete this work:
* [x] Move all current `README.md` contents to appropriate pages
* [x] Enable a GitHub workflow to automatically deploy docs ([example](https://github.com/paularmstrong/build-tracker/blob/main/.github/workflows/push.yml#L44-L60))
* [x] Update colors
* [x] Update main homepage
* [x] Remove (or update) contents of `docs/blog`
* [x] Update `README.md` to only include minimal relevant information, point to the GitHub hosted docs pages
* [ ] Once deployed, apply for [Algolia DocSearch](https://v2.docusaurus.io/docs/search)

### CODE REVIEW

**Comment by blakeblackshear** [comment]
The readme is getting a bit unwieldy, so I think this would be great.

**Comment by paularmstrong** [comment]
Sounds great. I'll keep splitting up the README as best I can just to get things started.

I have one important question at the moment: Should we run it in [docs-only mode](https://v2.docusaurus.io/docs/docs-introduction/#docs-only-mode) to make things easier to get started? (this would remove the "pages" and "blog", as well as the home page) In the future, this can be un-done if desired.

**Comment by blakeblackshear** [comment]
Seems like a good idea. We can treat the readme more like a homepage.

**Comment by paularmstrong** [comment]
Current iteration is available at https://frigate.paularmstrong1.vercel.app/

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          node-version: 12.x
+      - run: npm install
+      - name: Build docs
+        run: npm run build
+      - name: Deploy documentation
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./docs/build
```
Comment (paularmstrong) [inline .github/workflows/push.yml:28]: I'm not 100% certain on whether this action will use the working-directory set on line 15 or if it is always relative to the root.

File: docs/docs/configuration/cameras.md [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+```
+
+### Zones
+
+Zones allow you to define a specific area of the frame and apply additional filters for object types so you can determine whether or not an object is within a particular area. Zones cannot have the same name as a camera. If desired, a single zone can include multiple cameras if you have multiple cameras covering the same area by configuring zones with the same name for each camera.
+
+During testing, `draw_zones` should be set in the config to draw the zone on the frames so you can adjust as needed. The zone line will increase in thickness when any object enters the zone.
+
+To create a zone, follow the same steps above for a "Motion mask", but use the section of the web UI for creating a zone instead.
+
```
Comment (paularmstrong) [inline docs/docs/configuration/cameras.md:80]: I took some liberties to update these two sections a little to point to the web UI

File: docs/docs/usage/web.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+id: web
+title: Web Interface
+---
+
+Frigate comes bundled with a simple web ui that supports the following:
+
+- Show cameras
+- Browse events
+- Mask helper
```
Comment (paularmstrong) [inline docs/docs/usage/web.md:10]: This should get expanded, but can happen separately as well.

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+module.exports = {
+  title: 'Frigate',
+  tagline: 'NVR With Realtime Object Detection for IP Cameras',
+  url: 'https://blakeblackshear.github.io/frigate',
```
Comment (paularmstrong) [inline docs/docusaurus.config.js:4]: Not sure if you're going to use the GitHub default or set a custom domain. I believe this should work as a default.

File: docs/static/img/logo.svg [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M130 446.5C131.6 459.3 145 468 137 470C129 472 94 406.5 86 378.5C78 350.5 73.5 319 75.5 301C77.4999 283 181 255 181 247.5C181 240 147.5 247 146 241C144.5 235 171.3 238.6 178.5 229C189.75 214 204 216.5 213 208.5C222 200.5 233 170 235 157C237 144 215 129 209 119C203 109 222 102 268 83C314 64 460 22 462 27C464 32 414 53 379 66C344 79 287 104 287 111C287 118 290 123.5 288 139.5C286 155.5 285.76 162.971 282 173.5C279.5 180.5 277 197 282 212C286 224 299 233 305 235C310 235.333 323.8 235.8 339 235C358 234 385 236 385 241C385 246 344 243 344 250C344 257 386 249 385 256C384 263 350 260 332 260C317.6 260 296.333 259.333 287 256L285 263C281.667 263 274.7 265 267.5 265C258.5 265 258 268 241.5 268C225 268 230 267 215 266C200 265 144 308 134 322C124 336 130 370 130 385.5C130 399.428 128 430.5 130 446.5Z" fill="black"/>
+</svg>
```
Comment (paularmstrong) [inline docs/static/img/logo.svg:3]: I custom-made these in Figma. Let me know if you'd like the source files shared. Just need the email address you have signed up in Figma or else a way to send you a link that's not out in the open.

**Pull request merged**